extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java

```diff
@@ -69,6 +69,15 @@
     @ConfigItem(defaultValue = "true")
     public boolean transformUnproxyableClasses;
 
+    /**
+     * If set to true, the build will fail if an annotation that would result in an interceptor being created (such as
```

Comment:
Formally, the build should fail if a private method that is neither an observer nor a producer and is annotated with an interceptor binding. The transactional example is OK and should follow ;-).